### ISSUE TITLE
Phantoms don't show up depending on how the file is opened

### ISSUE THREAD

**Issue by rchl** [jfcherng-sublime/ST-OpenUri#5]
I've always randomly seen the issue of phantoms not being created but now I found out when that happens.

Opening the file from the "Go to anywhere" panel doesn't create the phantoms.
Opening the file from the sidebar creates the phantoms.

https://user-images.githubusercontent.com/153197/121652690-32beb500-ca9c-11eb-831b-4f1f42973206.mov

**Comment by jfcherng** [comment]
It looks like a transient has a view ID (`view.id()`) *right after* it gets created. But it won't have a window (`view.window()`)... And after a second, it has a window...

```py
import sublime
import sublime_plugin


def print_view_info(view: sublime.View) -> None:
    view_id = view.id()
    window = view.window()
    window_id = window.id() if window else -1

    print(f'v_id: {view_id}, w_id: {window_id}')


class TransientViewWindowIdTest(sublime_plugin.ViewEventListener):
    def __init__(self, view: sublime.View) -> None:
        self.view = view

        # you can NOT see a window id
        print_view_info(view)

        # you can see a window id
        sublime.set_timeout_async(lambda: print_view_info(view))
```

**Comment by rchl** [comment]
Checking stuff from `__init__` of the event listener is kinda asking for it though. Can you not do it from `on_load`?

**Comment by jfcherng** [comment]
> Can you not do it from on_load?

The result remains the same. No window is attached.

```py
import sublime
import sublime_plugin


def print_view_info(view: sublime.View) -> None:
    view_id = view.id()
    window = view.window()
    window_id = window.id() if window else -1

    print("v_id: {}, w_id: {}".format(view_id, window_id))


class TransientViewWindowIdTest(sublime_plugin.ViewEventListener):
    def __init__(self, view: sublime.View) -> None:
        self.view = view

    def on_load(self) -> None:
        # you can NOT see a window id
        print_view_info(self.view)
        # you can see a window id
        sublime.set_timeout_async(lambda: print_view_info(self.view))
```

**Comment by rchl** [comment]
Hmm, yes. I see that now. Looks like an ST bug.

**Comment by jfcherng** [comment]
Fixed in 247857e

**Comment by jfcherng** [comment]
> Hmm, yes. I see that now. Looks like an ST bug.

Reported https://github.com/sublimehq/sublime_text/issues/4444
